### PR TITLE
Server defaults

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-7442
https://issues.jboss.org/browse/ISPN-7444
https://issues.jboss.org/browse/ISPN-7445
https://issues.jboss.org/browse/ISPN-7446

Some important changes to ensure that we ship leaner server configurations with optimal values.

Here is an example of what the final configuration looks like:

https://gist.github.com/tristantarrant/78505d3b85f432b402b7a5733ecb274e/revisions

### CODE REVIEW

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteConfigurationResource.java [hunk 1]
```diff
@@ -55,21 +57,21 @@
             .setAllowExpression(true)
             .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
             .setValidator(new EnumValidator<>(BackupStrategy.class, true, true))
-            .setDefaultValue(new ModelNode().set(BackupStrategy.ASYNC.name()))
+            .setDefaultValue(new ModelNode().set(BackupConfiguration.STRATEGY.getDefaultValue().name()))
```
Comment (wburns) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteConfigurationResource.java]: Good catch :+1:

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteConfigurationResource.java [hunk 2]
```diff
@@ -55,21 +57,21 @@
...
-            .setDefaultValue(new ModelNode().set(10000L))
+            .setDefaultValue(new ModelNode().set(BackupConfiguration.REPLICATION_TIMEOUT.getDefaultValue()))
             .build()
     ;
     static final SimpleAttributeDefinition ENABLED = new SimpleAttributeDefinitionBuilder(ModelKeys.ENABLED, ModelType.BOOLEAN, true)
             .setXmlName(Attribute.ENABLED.getLocalName())
             .setAllowExpression(true)
             .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
-            .setDefaultValue(new ModelNode().set(true))
+            .setDefaultValue(new ModelNode().set(BackupConfiguration.ENABLED.name()))
```
Comment (wburns) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteConfigurationResource.java]: Looks like this should be `getDefaultValue()` ?

File: server/integration/infinispan/src/main/resources/subsystem-templates/infinispan-core.xml [hunk 3]
```diff
@@ -29,27 +23,27 @@
             <transport lock-timeout="60000" />
             <global-state />
 
-            <distributed-cache-configuration name="transactional" mode="SYNC" start="EAGER">
+            <distributed-cache-configuration name="transactional" mode="SYNC">
```
Comment (wburns) [inline server/integration/infinispan/src/main/resources/subsystem-templates/infinispan-core.xml]: Seems like we can remove all of the `mode="SYNC"` text as well right?

Comment (tristantarrant) [inline server/integration/infinispan/src/main/resources/subsystem-templates/infinispan-core.xml]: No, the cache mode is a compulsory attribute on caches.

Comment (ryanemerson) [inline server/integration/infinispan/src/main/resources/subsystem-templates/infinispan-core.xml]: @tristantarrant Is it worth making the mode attribute optional in order to further simplify the config? With a default value of "SYNC" if no mode is specified.

Comment (tristantarrant) [inline server/integration/infinispan/src/main/resources/subsystem-templates/infinispan-core.xml]: Yes, this sounds like a good idea. Having a compulsory attribute has always irked me a bit, and SYNC as a default is definitely a good default.

**Comment by tristantarrant** [comment]
@ryanemerson @wburns added ISPN-7446

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/Mode.java [hunk 4]
```diff
@@ -32,7 +32,8 @@
     ASYNC(false),
     ;
     private final boolean sync;
-    private Mode(boolean sync) {
+
+    Mode(boolean sync) {
```
Comment (wburns) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/Mode.java]: Guessing you didn't mean to change this?

Comment (tristantarrant) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/Mode.java]: An enum constructor is private by definition, so I removed the redundant declaration

Comment (wburns) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/Mode.java]: Ah yes true :)

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteStateTransferConfigurationResource.java [hunk 5]
```diff
@@ -22,6 +22,7 @@
 
 package org.jboss.as.clustering.infinispan.subsystem;
 
+import org.infinispan.configuration.cache.XSiteStateTransferConfiguration;
 import org.infinispan.configuration.cache.XSiteStateTransferConfigurationBuilder;
```
Comment (pruivo) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/BackupSiteStateTransferConfigurationResource.java]: the static field in ```XSiteStateTransferConfigurationBuilder``` can be removed, and the import can be removed as well.

**Comment by tristantarrant** [comment]
Done @pruivo

**Comment by pruivo** [comment]
looks good. waiting for CI results :+1:

**Comment by tristantarrant** [comment]
No related test failures in the previous run: http://ci.infinispan.org/viewLog.html?buildId=49477&tab=buildResultsDiv&buildTypeId=bt9

**Pull request merged**